scri/bms_transformations.py

```diff
@@ -0,0 +1,829 @@
+        if not ('rotation' in output_order and\
+                'boost' in output_order):
+            raise ValueError('Not enough transformations')
+        
+        if self.order == output_order:
+            return self
+        else:
+            L_spin_matrix = Lorentz_to_spin_matrix(self)
+            L_reordered = spin_matrix_to_Lorentz(L_spin_matrix, output_order=output_order)
+            return L_reordered
+
+    def is_identity(self, verbose=False):
+        """Check if a BMS transformation is the identity element
+        """
+        rotation_is_identity = np.allclose(self.rotation.components, [1,0,0,0])
+        boost_is_identity = np.allclose(self.boost, [0,0,0])
+        if verbose:
+            print(f"rotation is identity: {rotation_is_identity}")
+            print(f"boost is identity: {boost_is_identity}")
+        return rotation_is_identity and boost_is_identity
+
+    def inverse(self, output_order=None):
+        """Compute the inverse of a Lorentz transformation.
+        """
+        if output_order is None:
+            output_order = self.order[::-1]
+        
+        L_spin_matrix = Lorentz_to_spin_matrix(self)
+        L_spin_matrix_inverse = np.linalg.inv(L_spin_matrix)
+        L_inverse = spin_matrix_to_Lorentz(L_spin_matrix_inverse, output_order=output_order)
+        return L_inverse
+
+    def __eq__(self, other):
+        rotation_eq = np.allclose(self.rotation.components, other.rotation.components)
+        boost_eq = np.allclose(self.boost, other.boost)
+        return rotation_eq and boost_eq
+            
+    def compose(self, other, output_order=['rotation','boost']):
+        """Compose two Lorentz transformations.
+        
+        Parameters
+        ----------
+        other: Lorentz_transformation
+            2nd Lorentz transformation to be applied.
+        output_order: list
+            Order in which rotation and boost should be applied.
+        """
+        L1_spin_matrix = Lorentz_to_spin_matrix(self)
+        L2_spin_matrix = Lorentz_to_spin_matrix(other)
+        
+        Ls_spin_matrix = np.linalg.multi_dot([L2_spin_matrix, L1_spin_matrix])
+    
+        return spin_matrix_to_Lorentz(Ls_spin_matrix, output_order=output_order)
+
+    def conjugate(self, other, output_order=['rotation','boost']):
+        """Compute the conjugate of two Lorentz transformations.
+        
+        Parameters
+        ----------
+        other: Lorentz_transformation
+            2nd Lorentz transformation to be applied.
+        output_order: list
+            Order in which rotation and boost should be applied.
+        """
+        L1_spin_matrix = Lorentz_to_spin_matrix(self)
+        L2_spin_matrix = Lorentz_to_spin_matrix(other)
+        
+        conjugate_spin_matrix = np.linalg.multi_dot([L2_spin_matrix, L1_spin_matrix])\
+            - np.linalg.multi_dot([L1_spin_matrix, L2_spin_matrix])
+    
+        return spin_matrix_to_Lorentz(conjugate_spin_matrix, output_order=output_order)
```

Comment:
This looks like it's computing the commutator of spin matrices --- is that really the same as one Lorentz transformation conjugated by a second? I thought that rather it would have been L2 . L1 . inv(L2) . The documentation should also say what exactly is meant by the conjugation of one Lorentz transformation by another.